frontend/src/components/pages/GlobalCodeReviewDashboard/FilterByNamesFilters.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from "react";
+import FilterByNames from "./FilterByNames";
+import Typography from "@material-ui/core/Typography";
+import Grid from "@material-ui/core/Grid";
+import Paper from "@material-ui/core/Paper";
+import { makeStyles } from "@material-ui/styles";
+
+const useStyles = makeStyles((theme) => {
+  return {
+    filterByNamesContainer: {
+      margin: 0,
+      width: "100%",
+    },
+    filterByNamesItem: {
+      maxHeight: "90vh",
+      overflowY: "scroll",
+    },
+    filterByNamesContainerHeading: {
+      position: "sticky",
+      top: -5,
+      padding: "10px",
+      backgroundColor: theme.palette.background.default,
+      zIndex: 2,
+    },
+    filterByNamesItemBody: {
+      padding: "0px 10px",
+    },
+  };
+});
+
+function filteByFilterGroup({ allFilters, filter }) {
+  if (filter) {
+    return allFilters.filter((card) =>
+      card.toLowerCase().includes(filter.toLowerCase())
+    );
+  }
+  return allFilters;
+}
+
+export default function FilterByNamesFilters({
+  filterIncludeTags,
+  filterExcludeTags,
+
+  filterIncludeFlavours,
+  filterExcludeFlavours,
+
+  handleChangeFlavourFilter,
+  handleChangeTagFilter,
+
+  allTagNames,
+  allFlavours,
+  allTeamNames,
+  filterIncludeAssigneeTeams,
+  handleChangeAssigneeTeamFilter,
+}) {
+  const classes = useStyles();
+
+  const [flavourSearchTerm, setFlavourSearchTerm] = useState("");
```

Comment:
This should be defined inside FilterByNames/index.jsx.

Reply:
Also, this component is mixing presentation and implementation logic. Generally state and handlers should be implemented in index files and presentation should be in presentation files.